src/main/java/com/github/martapietka/ppm2pbm/PpmToPbmConverter.java

```diff
@@ -1,43 +1,48 @@
 package com.github.martapietka.ppm2pbm;
 
 import java.io.*;
-import java.nio.charset.StandardCharsets;
 
-public class PpmToPbmConverter {
+public class PpmToPbmConverter extends PpmConverter {
 
-    public static void convert(InputStream inputStream, OutputStream outputStream) throws IOException {
+    private final int threshold;
+    private final RgbToGrayscaleConverter rgbToGrayscaleConverter;
 
-        byte[] header = inputStream.readNBytes(3);
-
-        if (!((header[0] == 0x50) && (header[1] == 0x36) && (header[2] <= 0x1F))) {
-            System.err.println("Invalid file format");
-            System.exit(1);
-        }
+    public PpmToPbmConverter(int threshold, RgbToGrayscaleConverter rgbToGrayscaleConverter) {
+        this.threshold = threshold;
+        this.rgbToGrayscaleConverter = rgbToGrayscaleConverter;
+    }
 
-        int commentStart = inputStream.read();
-        if (commentStart != 0x23) {
-            System.err.println("Missing comment");
-            System.exit(1);
-        }
+    @Override
+    public void printHeader (OutputStream outputStream) throws IOException {
+        byte[] p1Header = {0x50, 0x31, 0xA};
+        outputStream.write(p1Header);
+    }
 
-        while (inputStream.read() > 0x1F) {
-            // skip comment
+    @Override
+    public void printOutput(InputStream inputStream, OutputStream outputStream, Header header) throws IOException {
+
+        byte[] rgbArray;
+        int counter = 1;
+
+        while ((rgbArray = inputStream.readNBytes(3)).length > 0) {
```

Comment:
Ta pętla powtarza się w obu implementacjach, spróbuj tutaj też coś ujednolicić. Zauważ, że zasadnicza różnica między implementacjami sprowadza się do tego, w jaki sposób przekształcisz wartości r,g,b na wartość typu `int` która następnie trafia do strumienia wyjściowego.

Jedyny "haczyk" polega na obsłudze licznika i łamania wiersza, która to ma sens tylko dla formatu PBM i to w tamtej implementacji powinno być umieszczone.

Podpowiem, że rozwiązaniem może być np. metoda przyjmująca r,g,b i zwracająca tablicę bajtów do wypisania:
* jednoelementową w przypadku PGM
* dwuelementową w przypadku PBM (bajt + separator)

Będzie tu pewnie potrzebny jakiś element stanowości żeby gdzieś tam, w oderwaniu od tej pętli, zachować obsługę tego licznika 😉